contracts/Uniswap/UniswapAnchoredView.sol

```diff
@@ -140,52 +130,42 @@ contract UniswapAnchoredView is UniswapConfig {
 
     /**
-     * @notice Post open oracle reporter prices, and recalculate stored price by comparing to anchor
-     * @dev We let anyone pay to post anything, but only prices from configured reporter will be stored in the view.
-     * @param messages The messages to post to the oracle
-     * @param signatures The signatures for the corresponding messages
-     * @param symbols The symbols to compare to anchor for authoritative reading
-     */
-    function postPrices(bytes[] calldata messages, bytes[] calldata signatures, string[] calldata symbols) external {
-        require(messages.length == signatures.length, "messages and signatures must be 1:1");
-
-        // Save the prices
-        for (uint i = 0; i < messages.length; i++) {
-            priceData.put(messages[i], signatures[i]);
-        }
+    * @notice This is called by the reporter whenever a new price is posted on-chain
+    * @dev called by AccessControlledOffChainAggregator
+    * @param currentAnswer the price
+    * @return valid bool
+    */
+    function validate(uint256 previousRoundId,
+            int256 previousAnswer,
+            uint256 currentRoundId,
+            int256 currentAnswer) external override returns (bool valid) {
+        
+        require(currentAnswer >= 0, "current answer cannot be negative");
+        uint256 reporterPrice = uint256(currentAnswer);
+        TokenConfig memory config = getTokenConfigByReporter(msg.sender);
 
         uint ethPrice = fetchEthPrice();
-
-        // Try to update the view storage
-        for (uint i = 0; i < symbols.length; i++) {
-            postPriceInternal(symbols[i], ethPrice);
-        }
-    }
-
-    function postPriceInternal(string memory symbol, uint ethPrice) internal {
-        TokenConfig memory config = getTokenConfigBySymbol(symbol);
         require(config.priceSource == PriceSource.REPORTER, "only reporter prices get posted");
-
-        bytes32 symbolHash = keccak256(abi.encodePacked(symbol));
-        uint reporterPrice = priceData.getPrice(reporter, symbol);
         uint anchorPrice;
-        if (symbolHash == ethHash) {
+        if (config.symbolHash == ethHash) {
             anchorPrice = ethPrice;
         } else {
-            anchorPrice = fetchAnchorPrice(symbol, config, ethPrice);
+            anchorPrice = fetchAnchorPrice(config.symbolHash, config, ethPrice);
         }
 
-        if (reporterInvalidated) {
-            prices[symbolHash] = anchorPrice;
-            emit PriceUpdated(symbol, anchorPrice);
+        if (reporterInvalidated[msg.sender]) {
+            prices[config.symbolHash] = anchorPrice;
+            emit PriceUpdated(config.symbolHash, anchorPrice);
         } else if (isWithinAnchor(reporterPrice, anchorPrice)) {
-            prices[symbolHash] = reporterPrice;
-            emit PriceUpdated(symbol, reporterPrice);
+            prices[config.symbolHash] = reporterPrice;
+            emit PriceUpdated(config.symbolHash, reporterPrice);
+            valid = true;
```

Comment:
We don't use this return value, though it also doesn't hurt to have it here.

Reply:
I'm happy to remove, but it makes sense from a contextual perspective.